library/std/src/fs.rs

```diff
@@ -1353,6 +1360,25 @@ impl Seek for Arc<File> {
     }
 }
 
+#[unstable(feature = "dirfd", issue = "120426")]
+impl Dir {
+    /// Opens a file relative to this directory.
+    pub fn open<P: AsRef<Path>>(&self, path: P) -> io::Result<File> {
+        self.inner.open(path).map(|f| File { inner: f })
+    }
+    /// Opens a file relative to this directory with the specified options.
+    pub fn open_with<P: AsRef<Path>>(&self, path: P, opts: &OpenOptions) -> io::Result<File> {
+        self.inner.open_with(path, &opts.0).map(|f| File { inner: f })
+    }
```

Comment:
Could you please add examples here, as well as tests? Tests can go in `std/src/fs/tests.rs`.